### DISCUSSION TITLE
Does API optional query string override conflicting mandatory default query string?

### DISCUSSION THREAD

**Discussion by afshinpir** [openssl/openssl#24619] [Q&A] [answered]
Hi All,

Let assume that I have enable fips on default context with following API:
```
EVP_default_properties_enable_fips(NULL, 1);
```
which adds "fips=yes" in the default query string. Now if I create a key with following call:
```
EVP_PKEY_CTX_new_from_name(NULL, "RSA", "?fips=yes")
```
What will happens for query string? Will fips be mandatory or preferred?

Best Regards

**Comment by paulidale** [comment, marked as answer]
In this case, the `?fips=yes` will override the global `fips=yes`.

More generally a global `provider=fips` with a `?fips=yes` locally will mandate: `provider=fips, ?fips=yes`.
